content/test/integration/service/deployments/deployment-entity-overlap.spec.ts

```diff
@@ -0,0 +1,177 @@
+
+/**
+ * This test verifies some scenarios that could happen during scene deployments.
+ */
+loadStandaloneTestEnvironment()("Integration - Deployment with Entity Overlaps", (testEnv) => {
+  const P1 = "0,0";
+  const P2 = "0,1";
+  const P3 = "1,1";
+  let identity: Identity;
+  let E1: EntityCombo, E2: EntityCombo;
+
+  beforeEach(async () => {
+    identity = createIdentity();
+    E1 = await buildDeployData([P1, P2], {
+      type: EntityType.SCENE,
+      metadata: {
+        main: "main.js",
+        scene: {
+          base: P1,
+          parcels: [P1, P2]
+        }
+      },
+      identity
+    });
+  });
+
+  testCaseWithComponents(
+    testEnv,
+    "When new scene is deployed on overlapping parcels, then new deployment removes previous scenes from orphaned parcels",
+    async (components) => {
+      // make validators stub
+      makeNoopValidator(components);
+      makeNoopServerValidator(components);
+
+      E2 = await buildDeployDataAfterEntity(E1, [P2], {
+        type: EntityType.SCENE,
+        metadata: {
+          main: "main.js",
+          scene: {
+            base: P2,
+            parcels: [P2]
+          }
+        },
+        identity
+      });
+
+      // Deploy E1 on P1, P2
+      await deploy(components, E1);
+      await assertDeploymentsAre(components, E1);
+
+      // Deploy E2 on P2
+      await deploy(components, E2);
+      await assertDeploymentsAre(components, E2); // E1 should no longer be active
+    }
+  );
+
+  testCaseWithComponents(
+    testEnv,
+    "When scene is deployed, then server checks for permissions",
+    async (components) => {
+      // make validators stub
+      stub(components.validator, "validate")
+        .onFirstCall()
+        .resolves({ ok: true })
+        .onSecondCall()
+        .resolves({
+          ok: false, errors: [
+            `The provided Eth Address does not have access to the following parcel: (${P1})`,
+            `The provided Eth Address does not have access to the following parcel: (${P2})`
+          ]
+        });
+      makeNoopServerValidator(components);
+
+      E2 = await buildDeployDataAfterEntity(E1, [P1, P2], {
+        type: EntityType.SCENE,
+        metadata: {
+          main: "main.js",
+          scene: {
+            base: P1,
+            parcels: [P1, P2]
+          }
+        }
+      });
+
+      // Deploy E1 on P1, P2
+      await deploy(components, E1);
+      await assertDeploymentsAre(components, E1);
+
+      // Deploy E2 on P1
+      await expect(deploy(components, E2)).rejects.toThrow('The provided Eth Address does not have access to the following parcel: (0,0),The provided Eth Address does not have access to the following parcel: (0,1)');
+      await assertDeploymentsAre(components, E1); // E2 should have not been deployed
+    }
+  );
+
+  testCaseWithComponents(
+    testEnv,
+    "When parcel changes owner, then new deployment by new owner succeeds and removes previous scenes from orphaned parcels",
```

Comment:
isn't this the same test case as the first one?

Reply:
In a sense... yes. Except for the 2 scenes belong to different identities. But giving the permission validation is mocked, they are somewhat very similar. In this case the overlap is also different (as it uses on overlapping parcel + one different one).